src/intptrcast.rs

```diff
@@ -0,0 +1,22 @@
+use std::cell::RefCell;
+
+use rustc::mir::interpret::AllocId;
+
+pub type MemoryState = RefCell<GlobalState>;
+
+#[derive(Clone, Debug)]
+pub struct GlobalState {
+    /// This field is used as a map between the address of each allocation and its `AllocId`
+    pub int_to_ptr_map: Vec<(u64, AllocId)>,
```

Comment:
Please state the important inariants: `int_to_ptr_map` is sorted, and `next_base_addr` is larger than any address that was previously made part of a block.